consensus/core/src/authority_node.rs

```diff
@@ -31,6 +32,8 @@ where
     start_time: Instant,
     transactions_client: Arc<TransactionClient>,
     leader_timeout_handle: LeaderTimeoutTaskHandle,
+    // Keeps core thread running.
+    _core_thread_handle: CoreThreadHandle,
```

Comment:
Since you are adding this maybe you can call the `stop` method as part of the `AuthorityNode::stop`